app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java

```diff
@@ -130,15 +137,25 @@ protected void initializeWorld() {
         mBrowserWidget = new BrowserWidget(this, currentSession);
         mPermissionDelegate.setParentWidgetHandle(mBrowserWidget.getHandle());
 
+        // Settings Widget
+        mSettingsWidget = new SettingsWidget(VRBrowserActivity.this);
+        mSettingsWidget.hide();
```

Comment:
The hide method is calling updateWidget before the widget is added., you can just set visible to false by default in the placement.

I think that we shouldn't create the settings widget at the startup, and create it lazily the first time the user clicks the settings button. Even more because I think that the settings widget also precreates internal widget screens. Having all these created will consume more memory for the app even when a user never gets in settings

Reply:
Fixed to only create the settings widget when required